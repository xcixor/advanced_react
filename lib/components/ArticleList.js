import React from 'react';
import Article from './Article';

const ArticleList = (props) => {
  return (
    <div className="article-list">
      <h2 className="page-head">Article List</h2>
      {Object.values(props.articles).map(article =>
        <Article
          key={article.id}
          article={article}
        />
      )}
    </div>
  );
};

export default ArticleList;