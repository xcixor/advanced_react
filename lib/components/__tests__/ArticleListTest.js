import React from 'react';
import ArticleList from '../ArticleList';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow } from 'enzyme';
import ArticleContainer from '../Article';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe('ArticleList', ()=> {
  const testProps = {
    articles: {
      a: {'id': 'a'},
      b: {'id': 'b'}
    },
  };

  ArticleContainer.propTypes = {};

  it('renders correctly', ()=> {
    const wrapper = shallow(
      <ArticleList
        {...testProps}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(toJson(wrapper.find('.article-list').children()).length).toBe(3);
    const titleElement = toJson(wrapper.find('.page-head'));
    expect(titleElement.children[0]).toEqual('Article List');
  });
});