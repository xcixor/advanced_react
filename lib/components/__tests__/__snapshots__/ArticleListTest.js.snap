// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div
  className="article-list"
>
  <h2
    className="page-head"
  >
    Article List
  </h2>
  <ArticleContainer
    article={
      Object {
        "id": "a",
      }
    }
    key="a"
  />
  <ArticleContainer
    article={
      Object {
        "id": "b",
      }
    }
    key="b"
  />
</div>
`;
